sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_connection.py

```diff
@@ -826,6 +831,17 @@ def create_session(
         self._outgoing_endpoints[assigned_channel] = session
         return session
 
+    def _send_empty_frame(self):
+        if self._can_write():
+            self._outgoing_empty()
+        self._schedule_keep_alive()
+
+    def _schedule_keep_alive(self):
+        if self._keep_alive_interval > 0 and not self._shutdown:
+            self._keep_alive_timer = Timer(self._keep_alive_interval, self._send_empty_frame)
```

Comment:
as a Timer does the time have to be reactivated every time its run?

Reply:
perhaps even an opportunity that if connection had activity in the past x seconds, then no need to keep alive

Reply:
maybe the same vein as this I tried using get_remote_timeout, which sends out our emptyFrame() as well and it caused our reconnection tests to fail